### PR TITLE
Fix: Unquoted `url()` data URL values that contain `+`

### LINKED ISSUE #2597: Can't resolve `url(data:image/svg+xml)` properly.
### [Reproduction Link](https://stylus-lang.com/try.html#?code=.weui-icon-search%20%7B%20mask-image%3A%20url(data%3Aimage%2Fsvg%2Bxml%2C%3Csvg%20width%3D%2224%22%20height%3D%2224%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Cpath%20d%3D%22M16.31%2015.561l4.114%204.115-.848.848-4.123-4.123a7%207%200%2011.857-.84zM16.8%2011a5.8%205.8%200%2010-11.6%200%205.8%205.8%200%200011.6%200z%22%20fill-rule%3D%22evenodd%22%2F%3E%3C%2Fsvg%3E)%20%7D)

For this code
```styl
 .icon-search {
   mask-image: url("data:image/svg+xml,%3Csvg%20width%3D%2224%22%20height%3D%2224%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Cpath%20d%3D%22M16.31%2015.561l4.114%204.115-.848.848-4.123-4.123a7%207%200%2011.857-.84zM16.8%2011a5.8%205.8%200%2010-11.6%200%205.8%205.8%200%200011.6%200z%22%20fill-rule%3D%22evenodd%22%2F%3E%3C%2Fsvg%3E")
 }
```

If url is wrapped with `"`, Stylus works fine.
However, if remove the `"`, compiler throws error `ParseError: illegal unary "%", missing left-hand operand`
```styl
 .icon-search {
   mask-image: url(data:image/svg+xml,%3Csvg%20width%3D%2224%22%20height%3D%2224%22%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%3E%3Cpath%20d%3D%22M16.31%2015.561l4.114%204.115-.848.848-4.123-4.123a7%207%200%2011.857-.84zM16.8%2011a5.8%205.8%200%2010-11.6%200%205.8%205.8%200%200011.6%200z%22%20fill-rule%3D%22evenodd%22%2F%3E%3C%2Fsvg%3E)
 }
```

Here is the documentation of [css url()](https://developer.mozilla.org/zh-CN/docs/Web/CSS/url()) which supports such usage.

### PR DESCRIPTION
**Pull request by groenroos**
**What**: Closes #2597 

**Why**: A value of `url(data:image/svg+xml)` without `'` or `"` produces a parse error `illegal unary "%", missing left-hand operand` - however, unquoted style is acceptable CSS3 syntax, so it should work in Stylus as well.

**How**: Unlike the error message would suggest, the culprit is **not** the `%` in the URL, but instead it's the `+` in `svg+xml` - this makes Stylus think it's an expression, like in the `bifs.url` test:

https://github.com/stylus/stylus/blob/9cb7635af60bf918ed8c2a990efd251e2e825975/test/cases/bifs.url.styl#L10-L13

I modified the the `urlchars` method in the lexer to accept `+` as urlchars, but only when the current `str` begins with a `+` and letters (otherwise the BIF test above would fail). At this point of parsing, the `str` would look like `+xml,...` so it's a fair guess that we have been parsing a data URL mimetype.

However, this methodology isn't completely foolproof (i.e. it doesn't actually _check_ that we're parsing a `data:` URL specifically), and although all the tests do currently pass, there is a chance this may cause other unintended edge case regressions.

**Checklist**:

- [ ] Documentation
- [x] Unit Tests
- [x] Code complete

### CODE REVIEW

File: lib/lexer.js [hunk 1]
```diff
@@ -285,7 +285,7 @@ Lexer.prototype = {
   urlchars: function() {
     var captures;
     if (!this.isURL) return;
-    if (captures = /^[\/:@.;?&=*!,<>#%0-9]+/.exec(this.str)) {
+    if (captures = /^\+[a-z]+/.exec(this.str) ? /^[\/:@.;?&+=*!,<>#%0-9]+/.exec(this.str) : /^[\/:@.;?&=*!,<>#%0-9]+/.exec(this.str)) {
```
Comment (iChenLei) [inline lib/lexer.js:288]: Can we refact these code to be more clear ? (e.g add comments for Regex
> Thanks for your great contribution, i invite you to mangage stylus'issues and pull requests

Comment (groenroos) [reply lib/lexer.js:288]: I'll see what I can do in terms of refactor, regex always looks messy IMO 😄 - but sure, I can definitely add comments!
